### ISSUE TITLE
Teleport Buttons Wrong Position & Stuck on Cooldown

### ISSUE THREAD

**Issue by ItsGreyToday** [d-rafferty/lethal-tube-removal#17]
This will be a lengthy and detailed observation, but a much needed one, I really like this mod.

### Screenshots
Profile **with** LethalPipeRemoval:
![image](https://github.com/d-rafferty/lethal-tube-removal/assets/84604322/3e476d6a-c08d-41cb-8f73-21db90ac7b3c)

Profile **without** LethalPipeRemoval:
![image](https://github.com/d-rafferty/lethal-tube-removal/assets/84604322/9580c8b8-f19a-49c0-9837-ee169d94cd8e)

----

### Describe the bug
a. Teleport buttons are not in their proper position (this is true whether both v is false, only v1 is true, only v2 is true)

b. Teleport cooldown (both beam in and beam out) gets stuck in "[Cooling Down]" (also without the countdown/timer), having a "tp" terminal command mod and executing it will also say that the tp command cannot be executed as it is still on cooldown. This behavior is desynced for hosts and clients, sometimes it will be bugged for both, sometimes only for the host or client/s. The steps to reproduce found below will be detailing the behavior where teleporter is bugged for host but not for client, and inverse teleporter will be bugged for client but not for host. This means that if the teleporter is not bugged for that player, that player can still use it and other players can still see them normally use it, meaning the desync is only on the state of cooldown.

c. Error logged regarding teleporter even when all 3 teleport stuff options are set to false: 
```
[Error  : Unity Log] NullReferenceException: Object reference not set to an instance of an object
Stack trace:
LethalTubeRemoval.Patches.TubeRemovalPatch.TeleporterStuff () (at <39aebf8165004319ade4f190f64f7398>:IL_025C)
(wrapper dynamic-method) ShipTeleporter.DMD<ShipTeleporter::Awake>(ShipTeleporter)
UnityEngine.Object:Instantiate(GameObject, Vector3, Quaternion, Transform)
StartOfRound:SpawnUnlockable(Int32)
StartOfRound:DMD<StartOfRound::LoadUnlockables>(StartOfRound)
StartOfRound:DMD<StartOfRound::Start>(StartOfRound)
MonoMod.Utils.DynamicMethodDefinition:Trampoline<StartOfRound::Start>?-1834049682(StartOfRound)
LethalLib.Modules.Items:StartOfRound_Start(orig_Start, StartOfRound)
MonoMod.Utils.DynamicMethodDefinition:Hook<StartOfRound::Start>?-333580496(StartOfRound)
MonoMod.Utils.DynamicMethodDefinition:Trampoline<StartOfRound::Start>?-934592246(StartOfRound)
LethalLib.Modules.Utilities:StartOfRound_Start(orig_Start, StartOfRound)
MonoMod.Utils.DynamicMethodDefinition:Hook<StartOfRound::Start>?1215659012(StartOfRound)
```

----

### To Reproduce
Steps to reproduce the behavior of bug b:

1. host lobby
2. client a joins lobby
3. make sure to have teleporter upgrade/s bought
4. host uses red teleporter -- at this point, red teleporter button should be bugged for host (happens even if only the host is in the lobby), i don't remember if teleporter will still be fine for client a, but for client b (which will join after this event) it still should be usable.
5. client b joins lobby
6. host lands on moon
7. host uses yellow (inverse) teleporter -- at this point, yellow teleporter button should be bugged for client/s (i have not tested if for all) and still be usable (many times) for host.

----

### Additional context
• UnlockableDefaults mod tested disabled/enabled, no effect on wrong tp button positions.

• Bug b has only been tested a few times. The observation of "when" clients/host gets bugged tp buttons may still vary.
• As LethalLib was mentioned in bug c, the bug has been tested and observed in LethalLib v0.14.2, v0.14.3, v0.14.4. Meaning LethalLib version shouldn't matter.
• From what I know, there are 3 mods in my profile code that does something with the teleporter:

1. GeneralImprovements (sets cd time)
2. BetterTeleporters (sets cd time, highest priority regarding kept items when teleporting)
3. LateGameUpgrades (has the lowest priority regarding teleporters, therefore low chance of being related to the bug)

• I don't remember if it is GeneralImprovements or BetterTeleporters that has the priority regarding the cooldown time.

• None of the 3 mentioned bugs appear when LethalPipeRemoval is disabled.
----

### Add a list of all of your mods
If needed, I have included the profile code below.

**r2modman code**: `018e63c3-808c-b7d5-a6a3-e35e572b6973`

Please note, profile code contains AdvancedCompany and AdvancedLevelLoader, it won't get downloaded if you do not use the forked r2modman, but that shouldn't matter as the purpose of the code is to see what mods are included and the configs that are set.

**Comment by ItsGreyToday** [comment]
> a. Teleport buttons are not in their proper position (this is true whether both v is false, only v1 is true, only v2 is true)

I forgot to include screenshots of v1 and v2 not being in their intended placement. So I went ahead to test them and somehow both v1 and v2 are now in the intended placement (wrong placement prob in older version).

TLDR: Teleport buttons are not in their proper position only for when v is false (by proper placement meaning the expected placement is the vanilla)

**Comment by d-rafferty** [comment]
Hm. Okay I had a similar issue with another mod I have been developing so I have a good idea of where to look. Thank you so much for the detailed description. I'll work on this and update back tomorrow with any progress.

**Comment by d-rafferty** [comment]
I have reproduced this and narrowed down the cause. Looking for a solution at the moment and will update soon.

**Comment by d-rafferty** [comment]
I was only able to do very limited network testing, but this appears to be fixed for now. If you have a large lobby, I encourage you to test this out before beginning any important runs as I could not try out max player size. Please report back if you have any bugs and I will reopen this issue.  Details for changes can be found here #18 

And once again thank you for the in-depth reporting it made it very easy to reproduce and isolate the (hopefully only) issue!

**Comment by d-rafferty** [comment]
This is live on thunderstore with v1.6.0
